### PR TITLE
Update star tracker noise model

### PR DESCRIPTION
**Pull request by galensavidge**
Baseline sim runs and looks the same as far as I can tell. The unit tests pass. Here's the test output with `show_plots = True`:

image

### CODE REVIEW

**Comment by galensavidge** [comment]
I changed the noise model from a vector add to an MRP add. I think that should give the proper around/cross/cross boresight error in ST Case frame. I'm happy to explain the reasoning behind doing that.

**Comment by galensavidge** [comment]
Also, right now the noise parameter is a covariance matrix but the off-diagonal elements aren't used at all. Do we want to change that @alboehm?

**Comment by alboehm** [comment]
Diagonal covariance sounds fine to me and to this guy https://www.researchgate.net/publication/252670817_Enhancing_attitude_estimation_accuracy_via_system_noise_optimization

**Comment by galensavidge** [comment]
> Diagonal covariance sounds fine to me and to this guy https://www.researchgate.net/publication/252670817_Enhancing_attitude_estimation_accuracy_via_system_noise_optimization

Well, just to be clear, I mean if you *do* set a non-diagonal covariance matrix the model will only use the diagonal terms. So we could either update it to sample from the whole covariance matrix (I'd need to review how to do that) or update the input type to be a list of variances instead of a covariance matrix. Or we could leave it as is.

File: aces/ExternalModules/starTrackerSensor/_UnitTest/test_starTrackerSensor.py [hunk 1]
```diff
@@ -6,35 +6,38 @@
...
 from Basilisk.utilities import unitTestSupport
 from Basilisk.architecture import messaging
 from Basilisk.utilities import macros
 from Basilisk.ExternalModules import starTrackerSensor
-import numpy as np
 from Basilisk.utilities import RigidBodyKinematics
 
-def test_starTrackerSensor(accuracy):
+def test_starTrackerSensor(accuracy, show_plots):
     r"""
```
Comment (alboehm) [inline aces/ExternalModules/starTrackerSensor/_UnitTest/test_starTrackerSensor.py:22]: Is that r a typo? I have it on my develop branch too

File: aces/ExternalModules/starTrackerSensor/starTrackerSensor.cpp [hunk 2]
```diff
@@ -10,6 +10,7 @@
 StarTrackerSensor::StarTrackerSensor()
 {
     this->rngSeed = 0x1badcad1;
+    this->randNumGen.seed(this->rngSeed);
```
Comment (alboehm) [inline aces/ExternalModules/starTrackerSensor/starTrackerSensor.cpp:13]: confused here. was this not initialized before? How did they use it?

**Comment by alboehm** [comment]
> Well, just to be clear, I mean if you _do_ set a non-diagonal covariance matrix the model will only use the diagonal terms. So we could either update it to sample from the whole covariance matrix (I'd need to review how to do that) or update the input type to be a list of variances instead of a covariance matrix. Or we could leave it as is.

Taking the sqrt of the diagonal is the way to sample from the whole covariance matrix. The off-diagonals are correlation terms that are helpful for the KF, but don't have to do with noise at any one time.

File: aces/ExternalModules/starTrackerSensor/_UnitTest/test_starTrackerSensor.py (hunk 1 above)
Comment (galensavidge) [reply aces/ExternalModules/starTrackerSensor/_UnitTest/test_starTrackerSensor.py:22]: I think that makes it a raw string (no escape characters) but I don't think it'll do anything here

File: aces/ExternalModules/starTrackerSensor/starTrackerSensor.cpp (hunk 2 above)
Comment (galensavidge) [reply aces/ExternalModules/starTrackerSensor/starTrackerSensor.cpp:13]: No idea but it looks like `rngSeed` isn't referenced anywhere on develop right now

**Comment by galensavidge** [comment]
> Taking the sqrt of the diagonal is the way to sample from the whole covariance matrix. The off-diagonals are correlation terms that are helpful for the KF, but don't have to do with noise at any one time. So, this is good.

Won't that give you noise that's uncorrelated, making the off-diagonal terms zero?

**Comment by alboehm** [comment]
> Won't that give you noise that's uncorrelated, making the off-diagonal terms zero?

You're right. This is how you sample from a covariance matrix:
image

I think you should merge this commit and make that a TODO in the code, so if someone wants to use a non-diagonal noise matrix, they know they have to update the model.

**Review by alboehm — APPROVED** [review]
Make TODO for using measurement covariance matrices that have off diagonal elements. Also please add axis labels and titles on the unit test graphs.

**Pull request merged**